packages/rs-drive-abci/src/abci/handlers.rs

```diff
@@ -254,13 +254,40 @@ where
         let mut block_execution_context_guard =
             self.platform.block_execution_context.write().unwrap();
 
-        let mut new_round = false;
+        let mut drop_block_execution_context = false;
         if let Some(block_execution_context) = block_execution_context_guard.as_mut() {
             // We are already in a block
             // This only makes sense if we were the proposer unless we are at a future round
             if block_execution_context.block_state_info.round != (request.round as u32) {
                 // We were not the proposer, and we should process something new
-                new_round = true;
+                drop_block_execution_context = true;
+            } else if let Some(current_block_hash) =
+                block_execution_context.block_state_info.block_hash
+            {
+                // There is also the possibility that this block already came in, but tenderdash crashed
+                // Now tenderdash is sending it again
+                if let Some(proposal_info) = block_execution_context.proposer_results.as_ref() {
+                    // We were the proposer as well, so we have the result in cache
+                    return Ok(ResponseProcessProposal {
+                        status: proto::response_process_proposal::ProposalStatus::Accept.into(),
+                        app_hash: proposal_info.app_hash.clone(),
+                        tx_results: proposal_info.tx_results.clone(),
+                        consensus_param_updates: proposal_info.consensus_param_updates.clone(),
+                        validator_set_update: proposal_info.validator_set_update.clone(),
+                    });
+                }
+
+                if current_block_hash.as_slice() == request.hash {
+                    // We were not the proposer, just drop the execution context
+                    // Todo: (maybe) cache previous results
+                    drop_block_execution_context = true;
+                } else {
+                    // We are getting a different block hash for a block of the same round
```

Comment:
Won't it be possible that proposer for the same round take different transactions from mempool or changed block time? (for examples it's round 100)

Reply:
It's for same round, not for different